docs/beyond_basics/101-181-metalad.rst

```diff
@@ -4,10 +4,363 @@ Metadata-Management with MetaLad
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ cat << EOT > metadata-zen.json
+   {
+     "type": "file",
+     "path": "zen.png",
+     "dataset_id": "2d540a9d-2ef7-4b5f-8931-7c92f483f0c7",
+     "dataset_version": "19f2d98d758116d099d467260a5a71082b2c6a29",
+     "extractor_name": "Picture2Words",
+     "extractor_version": "0.1",
+     "extraction_parameter": {},
+     "extraction_time": 1675113291.1464975,
+     "agent_name": "Overworked CTO",
+     "agent_email": "closetoburnout@randomtechconsultancy.com",
+     "extracted_metadata": {
+     "description": "A lake with waterlilies in a front of snow covered mountains"
+	 }
+   }
+   EOT
+
+Then, we redirect the content of the file into the :command:`meta-add` command in the command line.
+The following call would add the metadata entry to the current dataset, ``cozy-screensavers``:
+
+
+.. runrecord:: _examples/DL-101-181-105
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad meta-add -d . - < metadata-zen.json
+
+.. find-out-more:: meta-add validity checks
+
+	When adding metadata for the first time, its not uncommon to run into errors.
+	Its quite easy, for example, to miss a comma or quotation mark when creating a JSON object by hand.
+	But there are also some internal checks that might be surprising.
+	If you want to add the metadata above to your own dataset, you should make sure to adjust the ``dataset_id`` to the ID of your own dataset, found via the command ``datalad configuration get datalad.dataset.id`` - otherwise you'll see an error [#f4]_, and likewise the ``dataset_version``.
+	And in case you'd supply the ``extraction_time`` as "this morning at 8AM" instead of a time stamp, the command will be unhappy as well.
+	In case an error occurs, make sure to read the error message, and turn the the commands' ``--help`` for insights about requirements you might have missed.
+
+After the metadata has been added, you can view it via the command :command:`meta-dump`.
+The simplest form of this command is ``meta-dump -r``, which will show all metadata that is stored in the dataset in the current directory.
+To get more specific metadata records, you can give a dataset-file-path-pattern to ``meta-dump``, much like an argument to ``ls``, that identifies :term:`dataset ID`, version and a file within the dataset.
+The two parts are separated by ``:``. The following line would just dump all metadata for ``zen.png``.
+
+.. runrecord:: _examples/DL-101-181-106
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad meta-dump -d . .:zen.png
+
+This could also be printed a bit more readable:
+
+.. runrecord:: _examples/DL-101-181-107
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad -f json_pp meta-dump -d . .:zen.png
+
+.. find-out-more:: More complex metadata-dumps
+
+   TODO: add complex Dataset-file-path-pattern examples, e.g., with UUIDs, versions, etc
+
```

Comment:
Does anyone have complex dumps from the real-world? The docstring of meta-add contains the spec for this notation, but it has a complex-ish feel to it, so the more examples, the better :)